katello/restore.rb

```diff
@@ -82,11 +83,19 @@ def reset_katello
     def restore_psql_dumps
       puts "Restoring postgres dump files"
       run_cmd("katello-service start --only postgresql")
-      run_cmd("runuser - postgres -c 'dropdb foreman'")
-      run_cmd("runuser - postgres -c 'dropdb candlepin'")
-      run_cmd("runuser - postgres -c 'psql -f #{File.join(@dir, "pg_globals.dump")} postgres 2>/dev/null'")
-      run_cmd("runuser - postgres -c 'pg_restore -C -d postgres #{File.join(@dir, "foreman.dump")}'")
-      run_cmd("runuser - postgres -c 'pg_restore -C -d postgres #{File.join(@dir, "candlepin.dump")}'")
+      if db_config.any_remote_db?
+        if db_config.any_local_db?
+          run_cmd("runuser - postgres -c 'psql -f #{File.join(@dir, "pg_globals.dump")} postgres 2>/dev/null'")
+        end
+        run_cmd(db_config.pg_command(db_config.foreman, 'pg_restore', "--no-privileges --clean -n public #{File.join(@dir, "foreman.dump")}"))
+        run_cmd(db_config.pg_command(db_config.candlepin, 'pg_restore', "--no-privileges --clean -n public #{File.join(@dir, "candlepin.dump")}"))
+      else
+        run_cmd("runuser - postgres -c 'dropdb foreman'")
+        run_cmd("runuser - postgres -c 'dropdb candlepin'")
+        run_cmd("runuser - postgres -c 'psql -f #{File.join(@dir, "pg_globals.dump")} postgres 2>/dev/null'")
+        run_cmd("runuser - postgres -c 'pg_restore -C -d postgres #{File.join(@dir, "foreman.dump")}'")
+        run_cmd("runuser - postgres -c 'pg_restore -C -d postgres #{File.join(@dir, "candlepin.dump")}'")
```

Comment:
nitpick. Can these work in a similar way to `backup.rb`? Making use of the methods in `DBConfig`, just for consistency. 

https://github.com/Katello/katello-packaging/pull/499/files#diff-e5c9a75a6cce52c183d1d26a2500311eR219

Reply:
@sean797, I didn't want to touch the restore of the local DB as it is not functionally equivalent to the commands run as pg admin. However the difference is small and should affect non-standard foreman DB (custom schema changes, privileges etc.).  If you think it is safe we can remove the whole else part and use the same procedure that is used for remote DB. What do you think?

Reply:
Makes sense 👍